### DISCUSSION TITLE
Is there a way to determine a field's visibility status?

### DISCUSSION THREAD

**Discussion by iwilson001** [Hopding/pdf-lib#1644] [Q&A] [answered]
I am currently working with this library and I have some radio groups that are Yes/No except when no is selected, the field has a visibility of hidden so it should not show as checked. I've been combing through the acroField properties, but I can't seem to find it anywhere.
For reference, when viewing my PDF in Firefox, I can see it in the styles attribute like so:
![image](https://github.com/Hopding/pdf-lib/assets/61028059/b539282a-d469-44f0-abdb-b8dfc9571968)
Any help would be appreciated!

I am working with a pdf form. The form is filled out in a separate UI and then I am populating it with the values set by the user, using this package.

**Comment by iwilson001** [comment, marked as answer]
I didn't figure this out on my own. I got help and this is what we ended up with. Once we have the field, we can access the acroField property and subsequently, the getWidgets() function. From there it's a matter of iterating over the widgets and checking to see if it has the annotation flag of hidden `w.hasFlag(AnnotationFlags.Hidden)` where w is of type PDFWidgetAnnotation, and AnnotationFlags comes from this library, src/core/annotation/flags.ts.

**Reply by sepsol** [reply]
@iwilson001 You just saved my life! You don't know how many ChatGPT hallucinations I went through before I decided to Google the issue and came across this.

**Reply by imsahsol** [reply]
Not only does @iwilson001 not need anyone to find answers to his own questions, but he also enlightens us all by sharing his discoveries with the world. Not all heroes wear capes.
